template/core/data/src/main/java/io/bloco/core/data/network/OpenLibraryHttpClient.kt

```diff
@@ -0,0 +1,57 @@
+package io.bloco.core.data.network
+
+import io.bloco.core.commons.logDebug
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.plugins.DefaultRequest
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.plugins.logging.LogLevel
+import io.ktor.client.plugins.logging.Logger
+import io.ktor.client.plugins.logging.Logging
+import io.ktor.client.plugins.observer.ResponseObserver
+import io.ktor.client.request.header
+import io.ktor.http.ContentType
+import io.ktor.http.HttpHeaders
+import io.ktor.http.URLProtocol
+import io.ktor.serialization.kotlinx.json.json
+import kotlinx.serialization.json.Json
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class OpenLibraryHttpClient @Inject constructor() {
+
+    private val client by lazy {
+        HttpClient(CIO) {
+            expectSuccess = true
+
+            install(ContentNegotiation) {
+                json(Json {
+                    isLenient = true
+                    ignoreUnknownKeys = true
+                    allowSpecialFloatingPointValues = true
+                    useArrayPolymorphism = true
+                })
+            }
+
+            install(Logging) {
+                logger = object : Logger {
+                    override fun log(message: String) {
+                        logDebug { "Response: $message" }
+                    }
+                }
+                level = LogLevel.ALL
```

Comment:
This is not a good default for a release, we should lower the level for release builds.